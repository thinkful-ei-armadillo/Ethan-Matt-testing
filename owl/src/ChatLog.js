import React from 'react';
import Message from './Message';
import cuid from 'cuid';

function ChatLog (props) {

  const ChatLog = props.messages.map((u) => {

    return (
    <Message
      id = {u.participant.id}
      name={u.participant.name}
      avatar={u.participant.avatar}
      timestamp = {u.timestamp}
      type = {u.type}
      message ={u.message}
      key =  {cuid()}
    /> )
  });

  return (
    <div>
      {ChatLog}
    </div>
  );
}

export default ChatLog;
