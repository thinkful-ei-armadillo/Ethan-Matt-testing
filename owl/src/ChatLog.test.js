import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatLog from './ChatLog';

const messages = [
  {
      participant: {
        id: 3,
        name: 'Axe Kubicka',
        avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
        inSession: false,
        onStage: false
    },

      type: 'message',
      message: 'Hello world',
      time: 1548852646559,
      timestamp: 1548852484247
  },
  {
      participant: {
        id: 2,
        name: 'Caty Flucker',
        avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
      },
      type: 'thumbs-up',
      timestamp: 1548852484247
  },
  {
      participant: {
        id: 4,
        name: 'Frank Runciman',
        avatar:
            'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
    },
      type: 'thumbs-down',
      timestamp: 1548852484247
  },
  {
      participant: {
      id: 1,
        name: 'Koren Templeton',
        avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
      },
      type: 'raise-hand',
      timestamp: 1548852544247
  },
  {
      participant: {
      id: 1,
        name: 'Koren Templeton',
        avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
      },
      type: 'clap',
      timestamp: 1548852544247
  },
  {
      participant: {
      id: 1,
        name: 'Koren Templeton',
        avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
      },
      type: 'join',
      timestamp: 1548852544247
  },
  {
      participant: {
        id: 5,
        name: 'Ashla Attwool',
        avatar:
            'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    },
      type: 'leave',
      timestamp: 1548852604247
  },
  {
      participant: {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
      type: 'join-stage',
      timestamp: 1548852664247
  },
  {
      participant: {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
      type: 'leave-stage',
      timestamp: 1548852724247
  }
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatLog messages={messages}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<ChatLog messages={messages}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
