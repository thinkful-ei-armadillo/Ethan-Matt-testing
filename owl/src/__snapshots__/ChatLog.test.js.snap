// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <div
    className="system-message"
  >
    <div>
       
      <img
        alt="some avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <strong>
        Axe Kubicka
         
      </strong>
       
      1548852484247
    </div>
    <div>
      Hello world
    </div>
  </div>
  <div
    className="system-message"
  >
    <div>
       
      <img
        alt="some avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <strong>
        Caty Flucker
         
      </strong>
       
      1548852484247
    </div>
    <div />
  </div>
  <div
    className="system-message"
  >
    <div>
       
      <img
        alt="some avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <strong>
        Frank Runciman
         
      </strong>
       
      1548852484247
    </div>
    <div />
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Koren Templeton
         
      </strong>
      raised their hand
    </p>
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Koren Templeton
         
      </strong>
      clapped
    </p>
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Koren Templeton
         
      </strong>
      joined
    </p>
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Ashla Attwool
         
      </strong>
      left
    </p>
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Axe Kubicka
         
      </strong>
      joined the stage
    </p>
  </div>
  <div
    className="system-message"
  >
    <p>
      <strong>
        Axe Kubicka
         
      </strong>
      left the stage
    </p>
  </div>
</div>
`;
